components/org.wso2.carbon.identity.outbound.provisioning.connector.office365/src/main/java/org/wso2/carbon/identity/outbound/provisioning/connector/office365/Office365ProvisioningConnector.java

```diff
@@ -172,15 +173,62 @@ protected String createUser(ProvisioningEntity provisioningEntity) throws Identi
                 log.debug("Returning provisioned user's ID: " + provisionedId);
             }
         } catch (IOException e) {
-            log.error("Error while closing HttpClient.");
+            log.error("Error while closing HttpClient.", e);
         }
         return provisionedId;
     }
 
-    protected void updateUser(ProvisioningEntity provisioningEntity) {
+    /**
+     * Call the update user endpoint of Azure AD and update the user properties.
+     *
+     * @param provisioningEntity user to be provisioned
+     * @throws IdentityProvisioningException if the user can not be created in the Azure AD
+     */
+    protected void updateUser(ProvisioningEntity provisioningEntity) throws IdentityProvisioningException {
+
+        String provisionedUserId = provisioningEntity.getIdentifier().getIdentifier();
+
+        try (CloseableHttpClient httpclient = HttpClientBuilder.create().useSystemProperties().build()) {
 
-        log.warn("Update user is not implemented.");
-        // TODO: 8/14/18 Implement update user logic
+            JSONObject user = buildUserAsJson(provisioningEntity);
+            String updateUserEndpoint = Office365ConnectorConstants.OFFICE365_USER_ENDPOINT + '/' + provisionedUserId;
+            HttpPatch patchRequest = new HttpPatch(updateUserEndpoint);
+            setAuthorizationHeader(patchRequest);
+
+            StringEntity requestBody = new StringEntity(user.toString());
+            requestBody.setContentType(Office365ConnectorConstants.CONTENT_TYPE_APPLICATION_JSON);
+            patchRequest.setEntity(requestBody);
+            patchRequest.setHeader(Office365ConnectorConstants.CONTENT_TYPE, Office365ConnectorConstants
+                    .CONTENT_TYPE_APPLICATION_JSON);
+
+            try (CloseableHttpResponse response = httpclient.execute(patchRequest)) {
+
+                if (response.getStatusLine().getStatusCode() == HttpStatus.SC_NO_CONTENT) {
+                    if (log.isDebugEnabled()) {
+                        log.debug("Successfully updated the provisioned user with id " + provisionedUserId + " in " +
+                                "the Azure Active Directory.");
+                    }
+                } else {
+                    JSONObject jsonResponse = new JSONObject(new JSONTokener(new InputStreamReader(
+                            response.getEntity().getContent())));
+                    String errorMessage = jsonResponse.getJSONObject("error").getString("message");
```

Comment:
Can this be null?  Better handle null value
jsonResponse.getJSONObject("error").

Reply:
fixed